packages/frontend/app/api/cookieApi.ts

```diff
@@ -26,7 +26,31 @@ export async function fetchAuthToken(): Promise<string> {
 
     return result
   } catch (error) {
-    console.error('Failed to fetch auth token', error)
+    console.error('Failed to fetch auth token: ' + error)
     return ''
   }
 }
+
+export async function setQueueInviteCookie(
+  queueId: number,
+  courseId: number,
+  orgId: number,
+  courseInviteCode?: string,
+): Promise<void> {
+  try {
+    const cookieStore = cookies()
+    cookieStore.set(
+      'queueInviteInfo',
+      `${courseId},${queueId},${orgId},${courseInviteCode ? Buffer.from(courseInviteCode).toString('base64') : ''}`,
```

Comment:
since courseInviteCode can have special characters (and these characters are not URI/URL safe, which causing issues with encoding/decoding it), it needs to get encoded into base64 first, which will encode the special characters into a URI/URL safe format.